sample/src/main/java/org/smartregister/nativeform/domain/ConfigForm.java

```diff
@@ -0,0 +1,192 @@
+package org.smartregister.nativeform.domain;
+
+import android.content.Context;
+
+import com.vijay.jsonwizard.domain.Form;
+
+import org.smartregister.nativeform.util.FileReaderUtil;
+import org.smartregister.nativeform.util.ResourceType;
+
+import java.util.Set;
+
+public class ConfigForm {
```

Comment:
This class looks very similar to ` com.vijay.jsonwizard.domain.Form`.

Reply:
This was by design, its a wrapper class to enable serialization and deserialization of the Form class. The shortcoming with the form class is it has links to dynamic referenced constants 

It's converted back to form here
org.smartregister.nativeform.domain.ConfigForm#toForm

Reply:
Is there no way to share common fields using inheritance?

Reply:
Direct inheritance is not possible in this case. Unless the config object changes the names of the parameters, getters and setters will not work.  All these fields are changing types 

```

    private int homeAsUpIndicator;

    private int actionBarBackground;
    private int navigationBackground;

    private int backIcon;
```

to 

```
    private String actionBarBackground;
    private String navigationBackground;

    private String backIcon;

    private String nextLabel;
    private String previousLabel;
    private String saveLabel;

```

Reply:
Cool.